flow/connectors/s3/s3.go

```diff
@@ -117,41 +117,13 @@ func (c *S3Connector) ValidateCheck(ctx context.Context) error {
 		return fmt.Errorf("failed to delete from bucket: %w", delErr)
 	}
 
-	// check if we can ping external metadata
-	if c.pgMetadata != nil {
```

Comment:
This check isn't necessary; we ping connection before returning connection from pool